Bump Calico to 3.17 (3.17.1 implicitly)

consideRatio opened the pull request:
Just keeping it updated, not aware of any bug fix would be relevant or
not for any project relying on this action. The release of 3.17.0 was
made 23 Nov 2020, while 3.17.1 was made 10 Dec 2020. At this point I
think a bump would be safe from the initial bleeding edge issues.

## A calico-version input? I don't think so.
I considered if we want to have a calico-version input, but I think we don't. Then we would end up needing to keep track of version we support and if there are different versions that come with special quirks it would become really troublesome to grid-test k3s + calico versions etc. So, my thinking is that it makes sense to have this pinned.

---

I committed this change on top of #8 as it was modifying a common line. The PR by itself is really just 8094c17.

manics commented:
I agree on not making Calico configurable, we can treat it as an internal implementation detail of the K8s cluster. If someone needs an older version they can use an older tag of the role.

manics commented:
Want to rebase this? Then merge and release ` v0.2.3`?

consideRatio commented:
Rebased!

I figure it could make sense to bump the minor version because we added a feature, or perhaps even cutting 1.0.0 seems reasonable. Does the process of publishing a new version of the GitHub action equal pushing a tag btw?

manics reviewed (approved):
I normally treat `0.x.y` semver as `0.major.minor` but everyone has their own opinions on this so `0.3.0` is also good 😀. I've been creating releases using
https://github.com/jupyterhub/action-k3s-helm/releases/new
which lets you create a tag at the same time, but manually tagging and creating the release afterwards also works.

If we're aiming for `1.0.0` I'd like there to be a `1` alias that always points to the latest `1.x.y` tag in line with what many official actions do. I've got some ideas for doing that in a GitHub action that's triggered by a new tag but haven't had time.
